tdp/conftest.py

```diff
@@ -72,6 +68,7 @@ def db_engine(
     if request.param:
         init_database(engine)
     yield engine
+    BaseModel.metadata.drop_all(engine)
```

Comment:
We init the database only if `request.param` is `True`, so we must `drop_all` only if `request.param` is `True`.
```suggestion
    if request.param:
        BaseModel.metadata.drop_all(engine)
```

Reply:
done